formats/qcow-efi.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, modulesPath, ... }:
+{
+  # for virtio kernel drivers
+  imports = [
+    "${toString modulesPath}/profiles/qemu-guest.nix"
+  ];
+
+  options = {
+    diskSize = lib.mkOption {
+      default = "auto";
+      description = "The disk size in megabytes of the system disk image.";
+      type = with lib.types; oneOf [ ints.positive (enum [ "auto" ])];
+    };
+  };
+
+  config = {
+    fileSystems."/" = {
+      device = "/dev/disk/by-label/nixos";
+      autoResize = true;
+      fsType = "ext4";
+    };
+
+    fileSystems."/boot" = {
+      device = "/dev/disk/by-label/ESP";
+      fsType = "vfat";
+    };
+
+    boot.growPartition = true;
+    boot.kernelParams = [ "console=ttyS0" ];
```

Comment:
There are some other architecture specific console paths to consider: https://github.com/numtide/srvos/blob/6ee5c9e530453161d48b7c853ada2d9fefcf0017/common/serial.nix#L37

I think qemu on aarch64 for example uses: `ttyAMA0`

Reply:
Hm. Is this abstracted somewhere that we can reference, or should we copypasta the list from srvos?

Reply:
I think copy-paste is the most viable solution right now

Reply:
Addressed in [1424032](https://github.com/nix-community/nixos-generators/pull/204/commits/142403228301f0af5f1709a8b1c6790f948d53ee)